aws/resource_aws_elasticsearch_domain_test.go

```diff
@@ -1113,7 +1122,36 @@ resource "aws_elasticsearch_domain" "test" {
 }
 
 func testAccESDomainConfigWarm(rName, warmType string, enabled bool, warmCnt int) string {
-	return fmt.Sprintf(`
+	if enabled {
```

Comment:
can you flatten this to a single config with no if? if warm is disabled it will ignore the other params (you can put 0 and "")

Reply:
im taking this comment back. ill let maintainer decide on this. im split on what is the way to go. may be splitting this is the right way to go /:

Reply:
@gdavison can you give your input here?

Reply:
I tried it with a single config at first but the test failed. After applying the configuration the state file stores `warm_type` as `""` and `warm_count` as `0`, because as you said, those parameters are ignored during application when `warm_enabled` is false. The test then fails because a follow-up plan phase returns a non-empty diff. Yet, we cannot set `warm_type` or `warm_count` to zero values because we constrain them to specific values.

So the solution here is to not set those optional parameters.

Reply:
It looks like the only difference in the two configurations is that when `warm_enabled` is `true`, it also includes `warm_count` and `warm_type`. So you could do something like

```go
warm_config := ""
if enabled {
    warm_config = fmt.Sprintf(`
warm_count = %[1]d
warm_type. = %[2]q
`, warmCount, warmType)
}
```

In the config, then you could put
```go
fmt.Sprintf(`
...
cluster_config {
  ...
  warm_enabled = %t
  %s
  ...
}
...
`, rName, warmEnabled, warmConfig)
```

Reply:
that's certainly a lot less repetitive, thanks for the suggestion. I'll try to push an update before the cut off.